vhost-device-sound/src/device.rs

```diff
@@ -416,13 +361,16 @@ impl VhostUserSoundThread {
                 },
                 code
             );
-            desc_chain
-                .memory()
-                .write_obj(resp, desc_hdr.addr())
+            writter_status
+                .write_obj(resp)
                 .map_err(|_| Error::DescriptorWriteFailed)?;
-            used_len += desc_hdr.len();
 
-            if vring.add_used(desc_chain.head_index(), used_len).is_err() {
+            used_len += writter_status.bytes_written();
+
+            if vring
+                .add_used(desc_chain.head_index(), used_len as u32)
```

Comment:
I think we should at least log if used_len is overflow

Reply:
Do youo mean something like this?:
```rust
            if u32::try_from(used_len).is_err() {
                log::error!("used_len has overflowed");
            }
```

Reply:
Yep, but at that point should we at least pass `u32::MAX` to `add_used()`?

I'm thinking about something like this (maybe better to use `warn` if we don't propagate the error to the guest):

```
let used_len = match u32::try_from(used_len) {
    Ok(len) => len,
    Err(len) => {
        log::warn!("used_len {} overflows u32", len);
        u32::MAX
    }
};
```

Reply:
Yes, we should pass `u32:MAX` in that case. I will add that, thanks.

Reply:
I wonder if this should not go in another new commit just before the adding of the use of Reader/Writer.

Reply:
Yeah, it's pre-existing, so good point.

Reply:
I just realized that this is partially pre-existing. In the previous code, `used_len` is u32 and no casting is required (Although the previous code does not check for overflow during additions). When we start to use Reader/Writer, `used_len` becomes `usize` and casting is required. Overflow can happen during this conversation. I think it would be better to add this check to the commit that targets the control queue instead of using a new commit. If I create a new commit for this, I would have to change everything again when adding Reader/Writer to the control queue. What do you think?

Reply:
Thanks, sorry for the confusion, I will update the PR and we can discuss then.